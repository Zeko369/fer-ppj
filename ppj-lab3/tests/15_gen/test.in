<program>
 <lista_naredbi>
  <naredba>
   <naredba_pridruzivanja>
    IDN 1 v
    OP_PRIDRUZI 1 =
    <E>
     <T>
      <P>
       OP_MINUS 1 -
       <P>
        BROJ 1 358
      <T_lista>
       $
     <E_lista>
      OP_MINUS 1 -
      <E>
       <T>
        <P>
         BROJ 1 165
        <T_lista>
         $
       <E_lista>
        $
  <lista_naredbi>
   <naredba>
    <naredba_pridruzivanja>
     IDN 2 v
     OP_PRIDRUZI 2 =
     <E>
      <T>
       <P>
        OP_MINUS 2 -
        <P>
         BROJ 2 438
       <T_lista>
        OP_PUTA 2 *
        <T>
         <P>
          BROJ 2 74
         <T_lista>
          $
      <E_lista>
       $
   <lista_naredbi>
    <naredba>
     <naredba_pridruzivanja>
      IDN 3 v
      OP_PRIDRUZI 3 =
      <E>
       <T>
        <P>
         IDN 3 v
        <T_lista>
         $
       <E_lista>
        OP_MINUS 3 -
        <E>
         <T>
          <P>
           IDN 3 v
          <T_lista>
           $
         <E_lista>
          $
    <lista_naredbi>
     <naredba>
      <naredba_pridruzivanja>
       IDN 4 v
       OP_PRIDRUZI 4 =
       <E>
        <T>
         <P>
          BROJ 4 811
         <T_lista>
          $
        <E_lista>
         $
     <lista_naredbi>
      <naredba>
       <naredba_pridruzivanja>
        IDN 5 v
        OP_PRIDRUZI 5 =
        <E>
         <T>
          <P>
           BROJ 5 237
          <T_lista>
           $
         <E_lista>
          OP_MINUS 5 -
          <E>
           <T>
            <P>
             BROJ 5 459
            <T_lista>
             $
           <E_lista>
            $
      <lista_naredbi>
       <naredba>
        <naredba_pridruzivanja>
         IDN 6 v
         OP_PRIDRUZI 6 =
         <E>
          <T>
           <P>
            IDN 6 v
           <T_lista>
            $
          <E_lista>
           OP_MINUS 6 -
           <E>
            <T>
             <P>
              BROJ 6 1056
             <T_lista>
              OP_PUTA 6 *
              <T>
               <P>
                BROJ 6 331
               <T_lista>
                $
            <E_lista>
             $
       <lista_naredbi>
        <naredba>
         <naredba_pridruzivanja>
          IDN 7 n
          OP_PRIDRUZI 7 =
          <E>
           <T>
            <P>
             IDN 7 v
            <T_lista>
             $
           <E_lista>
            OP_MINUS 7 -
            <E>
             <T>
              <P>
               BROJ 7 86
              <T_lista>
               $
             <E_lista>
              $
        <lista_naredbi>
         <naredba>
          <naredba_pridruzivanja>
           IDN 8 v
           OP_PRIDRUZI 8 =
           <E>
            <T>
             <P>
              OP_PLUS 8 +
              <P>
               BROJ 8 219
             <T_lista>
              $
            <E_lista>
             $
         <lista_naredbi>
          <naredba>
           <za_petlja>
            KR_ZA 9 za
            IDN 9 z
            KR_OD 9 od
            <E>
             <T>
              <P>
               IDN 9 n
              <T_lista>
               OP_PUTA 9 *
               <T>
                <P>
                 BROJ 9 265
                <T_lista>
                 $
             <E_lista>
              OP_PLUS 9 +
              <E>
               <T>
                <P>
                 BROJ 9 129
                <T_lista>
                 $
               <E_lista>
                $
            KR_DO 9 do
            <E>
             <T>
              <P>
               L_ZAGRADA 9 (
               <E>
                <T>
                 <P>
                  BROJ 9 651
                 <T_lista>
                  OP_PUTA 9 *
                  <T>
                   <P>
                    BROJ 9 705
                   <T_lista>
                    OP_PUTA 9 *
                    <T>
                     <P>
                      BROJ 9 834
                     <T_lista>
                      $
                <E_lista>
                 $
               D_ZAGRADA 9 )
              <T_lista>
               OP_PUTA 9 *
               <T>
                <P>
                 BROJ 9 81
                <T_lista>
                 $
             <E_lista>
              OP_MINUS 9 -
              <E>
               <T>
                <P>
                 IDN 9 n
                <T_lista>
                 $
               <E_lista>
                OP_MINUS 9 -
                <E>
                 <T>
                  <P>
                   BROJ 9 760
                  <T_lista>
                   OP_PUTA 9 *
                   <T>
                    <P>
                     BROJ 9 191
                    <T_lista>
                     $
                 <E_lista>
                  $
            <lista_naredbi>
             <naredba>
              <naredba_pridruzivanja>
               IDN 10 z
               OP_PRIDRUZI 10 =
               <E>
                <T>
                 <P>
                  IDN 10 z
                 <T_lista>
                  $
                <E_lista>
                 OP_PLUS 10 +
                 <E>
                  <T>
                   <P>
                    BROJ 10 485
                   <T_lista>
                    OP_PUTA 10 *
                    <T>
                     <P>
                      BROJ 10 352
                     <T_lista>
                      $
                  <E_lista>
                   OP_PLUS 10 +
                   <E>
                    <T>
                     <P>
                      BROJ 10 42
                     <T_lista>
                      $
                    <E_lista>
                     $
             <lista_naredbi>
              <naredba>
               <naredba_pridruzivanja>
                IDN 11 v
                OP_PRIDRUZI 11 =
                <E>
                 <T>
                  <P>
                   L_ZAGRADA 11 (
                   <E>
                    <T>
                     <P>
                      BROJ 11 953
                     <T_lista>
                      $
                    <E_lista>
                     OP_PLUS 11 +
                     <E>
                      <T>
                       <P>
                        IDN 11 n
                       <T_lista>
                        $
                      <E_lista>
                       $
                   D_ZAGRADA 11 )
                  <T_lista>
                   $
                 <E_lista>
                  OP_MINUS 11 -
                  <E>
                   <T>
                    <P>
                     BROJ 11 369
                    <T_lista>
                     $
                   <E_lista>
                    $
              <lista_naredbi>
               <naredba>
                <naredba_pridruzivanja>
                 IDN 12 z
                 OP_PRIDRUZI 12 =
                 <E>
                  <T>
                   <P>
                    BROJ 12 397
                   <T_lista>
                    $
                  <E_lista>
                   OP_MINUS 12 -
                   <E>
                    <T>
                     <P>
                      BROJ 12 827
                     <T_lista>
                      $
                    <E_lista>
                     OP_PLUS 12 +
                     <E>
                      <T>
                       <P>
                        BROJ 12 1141
                       <T_lista>
                        $
                      <E_lista>
                       $
               <lista_naredbi>
                <naredba>
                 <za_petlja>
                  KR_ZA 13 za
                  IDN 13 n
                  KR_OD 13 od
                  <E>
                   <T>
                    <P>
                     L_ZAGRADA 13 (
                     <E>
                      <T>
                       <P>
                        L_ZAGRADA 13 (
                        <E>
                         <T>
                          <P>
                           OP_PLUS 13 +
                           <P>
                            BROJ 13 957
                          <T_lista>
                           $
                         <E_lista>
                          $
                        D_ZAGRADA 13 )
                       <T_lista>
                        $
                      <E_lista>
                       OP_PLUS 13 +
                       <E>
                        <T>
                         <P>
                          OP_PLUS 13 +
                          <P>
                           L_ZAGRADA 13 (
                           <E>
                            <T>
                             <P>
                              BROJ 13 757
                             <T_lista>
                              $
                            <E_lista>
                             OP_MINUS 13 -
                             <E>
                              <T>
                               <P>
                                OP_PLUS 13 +
                                <P>
                                 L_ZAGRADA 13 (
                                 <E>
                                  <T>
                                   <P>
                                    BROJ 13 800
                                   <T_lista>
                                    $
                                  <E_lista>
                                   $
                                 D_ZAGRADA 13 )
                               <T_lista>
                                $
                              <E_lista>
                               $
                           D_ZAGRADA 13 )
                         <T_lista>
                          $
                        <E_lista>
                         OP_MINUS 13 -
                         <E>
                          <T>
                           <P>
                            IDN 13 z
                           <T_lista>
                            OP_PUTA 13 *
                            <T>
                             <P>
                              IDN 13 n
                             <T_lista>
                              $
                          <E_lista>
                           $
                     D_ZAGRADA 13 )
                    <T_lista>
                     $
                   <E_lista>
                    OP_MINUS 13 -
                    <E>
                     <T>
                      <P>
                       IDN 13 v
                      <T_lista>
                       $
                     <E_lista>
                      $
                  KR_DO 13 do
                  <E>
                   <T>
                    <P>
                     L_ZAGRADA 13 (
                     <E>
                      <T>
                       <P>
                        IDN 13 v
                       <T_lista>
                        OP_PUTA 13 *
                        <T>
                         <P>
                          L_ZAGRADA 13 (
                          <E>
                           <T>
                            <P>
                             OP_MINUS 13 -
                             <P>
                              BROJ 13 1230
                            <T_lista>
                             $
                           <E_lista>
                            $
                          D_ZAGRADA 13 )
                         <T_lista>
                          $
                      <E_lista>
                       OP_MINUS 13 -
                       <E>
                        <T>
                         <P>
                          OP_PLUS 13 +
                          <P>
                           L_ZAGRADA 13 (
                           <E>
                            <T>
                             <P>
                              L_ZAGRADA 13 (
                              <E>
                               <T>
                                <P>
                                 BROJ 13 366
                                <T_lista>
                                 $
                               <E_lista>
                                OP_PLUS 13 +
                                <E>
                                 <T>
                                  <P>
                                   IDN 13 n
                                  <T_lista>
                                   $
                                 <E_lista>
                                  $
                              D_ZAGRADA 13 )
                             <T_lista>
                              $
                            <E_lista>
                             $
                           D_ZAGRADA 13 )
                         <T_lista>
                          $
                        <E_lista>
                         $
                     D_ZAGRADA 13 )
                    <T_lista>
                     $
                   <E_lista>
                    OP_MINUS 13 -
                    <E>
                     <T>
                      <P>
                       IDN 13 z
                      <T_lista>
                       $
                     <E_lista>
                      $
                  <lista_naredbi>
                   <naredba>
                    <naredba_pridruzivanja>
                     IDN 14 e
                     OP_PRIDRUZI 14 =
                     <E>
                      <T>
                       <P>
                        OP_PLUS 14 +
                        <P>
                         L_ZAGRADA 14 (
                         <E>
                          <T>
                           <P>
                            BROJ 14 51
                           <T_lista>
                            $
                          <E_lista>
                           $
                         D_ZAGRADA 14 )
                       <T_lista>
                        $
                      <E_lista>
                       $
                   <lista_naredbi>
                    $
                  KR_AZ 15 az
                <lista_naredbi>
                 $
            KR_AZ 16 az
          <lista_naredbi>
           <naredba>
            <naredba_pridruzivanja>
             IDN 17 n
             OP_PRIDRUZI 17 =
             <E>
              <T>
               <P>
                IDN 17 n
               <T_lista>
                $
              <E_lista>
               $
           <lista_naredbi>
            <naredba>
             <za_petlja>
              KR_ZA 18 za
              IDN 18 v
              KR_OD 18 od
              <E>
               <T>
                <P>
                 IDN 18 v
                <T_lista>
                 $
               <E_lista>
                OP_MINUS 18 -
                <E>
                 <T>
                  <P>
                   BROJ 18 767
                  <T_lista>
                   OP_PUTA 18 *
                   <T>
                    <P>
                     L_ZAGRADA 18 (
                     <E>
                      <T>
                       <P>
                        BROJ 18 463
                       <T_lista>
                        $
                      <E_lista>
                       $
                     D_ZAGRADA 18 )
                    <T_lista>
                     $
                 <E_lista>
                  $
              KR_DO 18 do
              <E>
               <T>
                <P>
                 IDN 18 v
                <T_lista>
                 OP_PUTA 18 *
                 <T>
                  <P>
                   BROJ 18 1033
                  <T_lista>
                   OP_PUTA 18 *
                   <T>
                    <P>
                     IDN 18 n
                    <T_lista>
                     $
               <E_lista>
                $
              <lista_naredbi>
               $
              KR_AZ 19 az
            <lista_naredbi>
             $
