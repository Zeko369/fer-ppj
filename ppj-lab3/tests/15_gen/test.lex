IDN 1 v
OP_PRIDRUZI 1 =
OP_MINUS 1 -
BROJ 1 358
OP_MINUS 1 -
BROJ 1 165
IDN 2 v
OP_PRIDRUZI 2 =
OP_MINUS 2 -
BROJ 2 438
OP_PUTA 2 *
BROJ 2 74
IDN 3 v
OP_PRIDRUZI 3 =
IDN 3 v
OP_MINUS 3 -
IDN 3 v
IDN 4 v
OP_PRIDRUZI 4 =
BROJ 4 811
IDN 5 v
OP_PRIDRUZI 5 =
BROJ 5 237
OP_MINUS 5 -
BROJ 5 459
IDN 6 v
OP_PRIDRUZI 6 =
IDN 6 v
OP_MINUS 6 -
BROJ 6 1056
OP_PUTA 6 *
BROJ 6 331
IDN 7 n
OP_PRIDRUZI 7 =
IDN 7 v
OP_MINUS 7 -
BROJ 7 86
IDN 8 v
OP_PRIDRUZI 8 =
OP_PLUS 8 +
BROJ 8 219
KR_ZA 9 za
IDN 9 z
KR_OD 9 od
IDN 9 n
OP_PUTA 9 *
BROJ 9 265
OP_PLUS 9 +
BROJ 9 129
KR_DO 9 do
L_ZAGRADA 9 (
BROJ 9 651
OP_PUTA 9 *
BROJ 9 705
OP_PUTA 9 *
BROJ 9 834
D_ZAGRADA 9 )
OP_PUTA 9 *
BROJ 9 81
OP_MINUS 9 -
IDN 9 n
OP_MINUS 9 -
BROJ 9 760
OP_PUTA 9 *
BROJ 9 191
IDN 10 z
OP_PRIDRUZI 10 =
IDN 10 z
OP_PLUS 10 +
BROJ 10 485
OP_PUTA 10 *
BROJ 10 352
OP_PLUS 10 +
BROJ 10 42
IDN 11 v
OP_PRIDRUZI 11 =
L_ZAGRADA 11 (
BROJ 11 953
OP_PLUS 11 +
IDN 11 n
D_ZAGRADA 11 )
OP_MINUS 11 -
BROJ 11 369
IDN 12 z
OP_PRIDRUZI 12 =
BROJ 12 397
OP_MINUS 12 -
BROJ 12 827
OP_PLUS 12 +
BROJ 12 1141
KR_ZA 13 za
IDN 13 n
KR_OD 13 od
L_ZAGRADA 13 (
L_ZAGRADA 13 (
OP_PLUS 13 +
BROJ 13 957
D_ZAGRADA 13 )
OP_PLUS 13 +
OP_PLUS 13 +
L_ZAGRADA 13 (
BROJ 13 757
OP_MINUS 13 -
OP_PLUS 13 +
L_ZAGRADA 13 (
BROJ 13 800
D_ZAGRADA 13 )
D_ZAGRADA 13 )
OP_MINUS 13 -
IDN 13 z
OP_PUTA 13 *
IDN 13 n
D_ZAGRADA 13 )
OP_MINUS 13 -
IDN 13 v
KR_DO 13 do
L_ZAGRADA 13 (
IDN 13 v
OP_PUTA 13 *
L_ZAGRADA 13 (
OP_MINUS 13 -
BROJ 13 1230
D_ZAGRADA 13 )
OP_MINUS 13 -
OP_PLUS 13 +
L_ZAGRADA 13 (
L_ZAGRADA 13 (
BROJ 13 366
OP_PLUS 13 +
IDN 13 n
D_ZAGRADA 13 )
D_ZAGRADA 13 )
D_ZAGRADA 13 )
OP_MINUS 13 -
IDN 13 z
IDN 14 e
OP_PRIDRUZI 14 =
OP_PLUS 14 +
L_ZAGRADA 14 (
BROJ 14 51
D_ZAGRADA 14 )
KR_AZ 15 az
KR_AZ 16 az
IDN 17 n
OP_PRIDRUZI 17 =
IDN 17 n
KR_ZA 18 za
IDN 18 v
KR_OD 18 od
IDN 18 v
OP_MINUS 18 -
BROJ 18 767
OP_PUTA 18 *
L_ZAGRADA 18 (
BROJ 18 463
D_ZAGRADA 18 )
KR_DO 18 do
IDN 18 v
OP_PUTA 18 *
BROJ 18 1033
OP_PUTA 18 *
IDN 18 n
KR_AZ 19 az
