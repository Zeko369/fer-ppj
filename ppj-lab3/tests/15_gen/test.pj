v = -358 - 165
v = -438*74
v = v - v
v = 811
v = 237 - 459
v = v - 1056*331
n = v - 86
v = +219
za z od n*265 + 129 do (651*705*834)*81 - n - 760*191
  z = z + 485*352 + 42
  v = (953 + n) - 369
  z = 397 - 827 + 1141
  za n od ((+957) + +(757 - +(800)) - z*n) - v do (v*(-1230) - +((366 + n))) - z
    e = +(51)
  az
az
n = n
za v od v - 767*(463) do v*1033*n
az
