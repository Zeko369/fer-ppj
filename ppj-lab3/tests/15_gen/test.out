3 1 v
3 1 v
6 1 v
7 1 v
9 7 n
9 7 n
10 9 z
11 7 n
13 9 z
err 13 n
